rust/agama-server/src/web/common/jobs.rs

```diff
@@ -0,0 +1,186 @@
+use std::{collections::HashMap, pin::Pin, task::Poll};
+
+use agama_lib::{
+    dbus::get_optional_property,
+    error::ServiceError,
+    jobs::{client::JobsClient, Job},
+    property_from_dbus,
+};
+use axum::{extract::State, routing::get, Json, Router};
+use futures_util::{ready, Stream};
+use pin_project::pin_project;
+use tokio::sync::mpsc::unbounded_channel;
+use tokio_stream::{wrappers::UnboundedReceiverStream, StreamExt};
+use zbus::zvariant::{ObjectPath, OwnedObjectPath, OwnedValue};
+
+use crate::{
+    dbus::{DBusObjectChange, DBusObjectChangesStream, ObjectsCache},
+    error::Error,
+    web::Event,
+};
+
+/// Builds a router for the jobs objects.
+pub async fn jobs_service<T>(
+    dbus: &zbus::Connection,
+    destination: &'static str,
+    path: &'static str,
+) -> Result<Router<T>, ServiceError> {
+    let client = JobsClient::new(dbus.clone(), destination, path).await?;
+    let state = JobsState { client };
+    Ok(Router::new().route("/jobs", get(jobs)).with_state(state))
+}
+
+#[derive(Clone)]
+struct JobsState<'a> {
+    client: JobsClient<'a>,
+}
+
+async fn jobs(State(state): State<JobsState<'_>>) -> Result<Json<Vec<Job>>, Error> {
```

Comment:
where is openAPI documentation for this route?

Reply:
It is an unsolved problem: the jobs can appear under different URLs, so we are not documenting them (check the [issues API](https://github.com/openSUSE/agama/blob/master/rust/agama-server/src/web/common.rs#L288)). Something to solve in the future.